### PR TITLE
Simple benchmark script

### PR DESCRIPTION
**Pull request by shashi**
```
Benchmarkresults:
    Package: /home/shashi/.julia/dev/SymbolicUtils
    Date: 2 May 2020 - 10:16
    Package commit: c62477
    Julia commit: 381693
    BenchmarkGroup:
        6-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "rule:noop:Sym" => Trial(34.001 ns)
          "rule:noop:Int" => Trial(21.444 ns)
          "ruleset:noop:Term" => Trial(2.823 μs)
          "ruleset:noop:Int" => Trial(0.024 ns)
          "ruleset:noop:Sym" => Trial(139.023 ns)
          "rule:noop:Term" => Trial(495.682 ns)
```

Will post comparison vs master soon. ImmutableDict is much faster it turns out.

### CODE REVIEW

**Comment by shashi** [comment]
```julia
julia> using PkgBenchmark

# I moved benchmark.jl outside this repository and asked PkgB to benchmark that
# on both master and current branch.
julia> j = judge(dirname(dirname(pathof(SymbolicUtils))), "master", retune=true, script="../benchmarks.jl")

julia> j.baseline_results # master
Benchmarkresults:
    Package: /home/shashi/.julia/dev/SymbolicUtils
    Date: 2 May 2020 - 15:32
    Package commit: ce8e05
    Julia commit: 381693
    BenchmarkGroup:
        1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "overhead" => 2-element BenchmarkTools.BenchmarkGroup:
                  tags: []
                  "ruleset" => 3-element BenchmarkTools.BenchmarkGroup:
                          tags: []
                          "noop:Int" => Trial(0.026 ns)
                          "noop:Sym" => Trial(798.879 ns)
                          "noop:Term" => Trial(3.802 μs)
                  "rule" => 3-element BenchmarkTools.BenchmarkGroup:
                          tags: []
                          "noop:Int" => Trial(667.937 ns)
                          "noop:Sym" => Trial(664.013 ns)
                          "noop:Term" => Trial(664.166 ns)

julia> j.target_results # this branch
Benchmarkresults:
    Package: /home/shashi/.julia/dev/SymbolicUtils
    Date: 2 May 2020 - 15:32
    Package commit: af55ca
    Julia commit: 381693
    BenchmarkGroup:
        1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "overhead" => 2-element BenchmarkTools.BenchmarkGroup:
                  tags: []
                  "ruleset" => 3-element BenchmarkTools.BenchmarkGroup:
                          tags: []
                          "noop:Int" => Trial(0.024 ns)
                          "noop:Sym" => Trial(77.371 ns)
                          "noop:Term" => Trial(1.857 μs)
                  "rule" => 3-element BenchmarkTools.BenchmarkGroup:
                          tags: []
                          "noop:Int" => Trial(12.334 ns)
                          "noop:Sym" => Trial(17.503 ns)
                          "noop:Term" => Trial(17.564 ns)
```

File: src/rule_dsl.jl [hunk 1]
```diff
@@ -227,7 +227,7 @@ struct RuleRewriteError
     expr
 end
 
-function (r::RuleSet)(@nospecialize(term); depth=typemax(Int), applyall=false, recurse=true)
+function (r::RuleSet)(term; depth=typemax(Int), applyall=false, recurse=true)
```
Comment (shashi) [inline src/rule_dsl.jl:236]: @MasonProtter check commit message :)

Comment (MasonProtter) [reply src/rule_dsl.jl:236]: Sorry, I don't understand.

**Comment by MasonProtter** [comment]
I'm not sure running benchmarks on travis is such a good idea. Travis runs things in VMs so resources can be given and taken away from your job unpredictably leading to very large amounts of noise.

File: src/rule_dsl.jl (hunk 1 above)
Comment (shashi) [reply src/rule_dsl.jl:236]:
```
Before

julia> @btime run(RuleSet([r]), a, 1000000)
  135.737 ms (3000003 allocations: 61.04 MiB)

After

julia> @btime run(RuleSet([r]), a, 1000000)
  82.478 ms (3000003 allocations: 61.04 MiB)
```

**Comment by shashi** [comment]
Yeah, this is just a first line of defense, if there's alarm we can check locally. This runs the most basic benchmarks for both master and PR branch. It also makes sure the benchmark keep working.

**Pull request merged**